app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
+      return self.subnet6.vlanid if self.subnet6
```

Comment:
What else should be returned if it's blank? `nil`? Previous behaviour probably resulted in an empty string too ...